5.2_pieceOfCake.py

```diff
@@ -1,13 +1,13 @@
-def calculate_recipe_price(recipe, **kwargs):
-    total_price = 0
-    optional_ingredients = []
-    # Check if optional argument is provided and assign it to optional_ingredients list.
-    for key, value in kwargs.items():
-        if key == "optional":
-            optional_ingredients = value
-    # Calculate the total price of the recipe.
-    for ingredient, percentage in kwargs.items():
-        if ingredient in recipe and ingredient not in optional_ingredients:
-            total_price += recipe[ingredient] * (percentage / 100)
-
+def calculate_recipe_price(recipe, **kwargs):
+    total_price = 0
+    optional_ingredients = []
+    # Check if optional argument is provided and assign it to optional_ingredients list.
+    for key, value in kwargs.items():

```

Comment:
Do you really need this loop?
Think of a way to combine this loop and the following loop in an efficient way.